prboom2/src/p_maputl.c

```diff
@@ -725,6 +725,195 @@ dboolean P_PathTraverse(fixed_t x1, fixed_t y1, fixed_t x2, fixed_t y2,
   return P_TraverseIntercepts(trav, FRACUNIT);
 }
 
+//
+// RoughBlockCheck
+// [XA] adapted from Hexen -- used by P_RoughMonsterSearch
+//
+
+static mobj_t *RoughBlockCheck(mobj_t *mo, int index, angle_t fov)
+{
+  mobj_t *link;
+  mobj_t *master;
+  angle_t angle, minang, maxang;
+
+  // pre-calculate fov check stuff since it
+  // won't change during the blocklinks loop
+  if (fov > 0) {
+    minang = mo->angle - fov / 2;
+    maxang = mo->angle + fov / 2;
+  }
+
+  link = blocklinks[index];
+  while (link)
+  {
+    // skip non-shootable actors
+    if (!(link->flags & MF_SHOOTABLE))
+    {
+      link = link->bnext;
+      continue;
+    }
+
+    // skip other players in co-op
+    if (netgame && !deathmatch && link->player)
```

Comment:
Isn't this redundant with the friend check? "Does the player have MF_FRIEND assigned in deathmatch?" is a question I don't know the answer to btw, which is likely important for the check below.

Reply:
I'm not seeing anywhere in code offhand that switches MF_FRIEND off in deathmatch... I wonder if that leads to more subtle bugs in places. O_O